File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/media/MediaRepository.java
package org.dvbviewer.controller.data.media;

import org.apache.commons.collections4.CollectionUtils;
import org.apache.commons.lang3.StringUtils;
import org.dvbviewer.controller.data.media.xml.Dir;
import org.dvbviewer.controller.data.media.xml.File;
import org.dvbviewer.controller.data.media.xml.VideoDirsFiles;
import org.dvbviewer.controller.entities.NaturalOrderComparator;
import org.dvbviewer.controller.io.api.APIClient;
import org.dvbviewer.controller.io.api.DMSInterface;

import java.io.IOException;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

/**
 * Created by rbaun on 02.04.18.
 */

public class MediaRepository {

    private static final String TAG = MediaRepository.class.getName();

    DMSInterface dmsInterface;

    List<MediaFile> medias;

    NaturalOrderComparator comparator = new NaturalOrderComparator();

    public MediaRepository() {
        dmsInterface = APIClient.getClient().create(DMSInterface.class);
    }

    public List<MediaFile> getMedias(long dirid) throws IOException {
            final VideoDirsFiles videoDirs = dmsInterface.getMediaDir(dirid).execute().body();
            final List<MediaFile> mediaFiles = new ArrayList<>();
            if (CollectionUtils.isNotEmpty(videoDirs.getDirs())) {
                final List<MediaFile> dirs = new ArrayList<>();
                for (Dir dir : videoDirs.getDirs()) {
                    MediaFile f = new MediaFile();
                    f.setDirId(dir.getDirid());
                    String[] dirArr = StringUtils.split(dir.getPath(), "\\");
                    int index = Math.max(0, dirArr.length-1);
                    f.setName(dirArr[index]);
                    dirs.add(f);
                }
                Collections.sort(dirs, comparator);
                mediaFiles.addAll(dirs);
            }
            if (CollectionUtils.isNotEmpty(videoDirs.getFiles())) {
                final List<MediaFile> files = new ArrayList<>();
                for (File file : videoDirs.getFiles()) {
                    MediaFile f = new MediaFile();
                    f.setDirId(-1l);
                    f.setId(file.getObjid());
                    f.setName(file.getName());
                    f.setThumb(file.getThumb());
                    files.add(f);
                }
                Collections.sort(files, comparator);
                mediaFiles.addAll(files);
            }
            return mediaFiles;
    }


}

File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/media/MediaRepository.kt
package org.dvbviewer.controller.data.media

import org.apache.commons.collections4.CollectionUtils
import org.apache.commons.lang3.StringUtils
import org.dvbviewer.controller.entities.NaturalOrderComparator
import org.dvbviewer.controller.io.api.DMSInterface
import java.util.*
import kotlin.math.max

/**
 * Created by rbaun on 02.04.18.
 */
class MediaRepository(private val dmsInterface: DMSInterface) {
    private val comparator = NaturalOrderComparator()

    fun getMedias(dirid: Long): List<MediaFile> {
        val videoDirs = dmsInterface.getMediaDir(dirid).execute().body()
        val mediaFiles = ArrayList<MediaFile>()
        if (CollectionUtils.isNotEmpty(videoDirs!!.dirs)) {
            val dirs = ArrayList<MediaFile>()
            videoDirs.dirs?.forEach { dir ->
                val f = MediaFile()
                f.dirId = dir.dirid
                val dirArr = StringUtils.split(dir.path, "\\")
                val index = max(0, dirArr.size - 1)
                f.name = dirArr[index]
                dirs.add(f)
            }
            Collections.sort(dirs, comparator)
            mediaFiles.addAll(dirs)
        }
        if (CollectionUtils.isNotEmpty(videoDirs.files)) {
            val files = ArrayList<MediaFile>()
            videoDirs.files?.forEach { file ->
                val f = MediaFile()
                f.dirId = -1L
                f.id = file.objid
                f.name = file.name
                f.thumb = file.thumb
                files.add(f)
            }
            Collections.sort(files, comparator)
            mediaFiles.addAll(files)
        }
        return mediaFiles
    }

}